src/main/java/crazypants/util/FluidUtil.java

```diff
@@ -227,47 +328,137 @@ public static ItemStack fillContainerFromInternalTank(ITankAccess tank, ItemStac
    */
   public static boolean fillPlayerHandItemFromInternalTank(World world, int x, int y, int z, EntityPlayer entityPlayer,
       ITankAccess tank) {
-    ItemStack item = entityPlayer.inventory.getCurrentItem();
-    if (item == null || item.getItem() == null) {
-      return false;
-    }
-    ItemStack filledContainer = fillContainerFromInternalTank(tank, item, !entityPlayer.capabilities.isCreativeMode);
-    if (filledContainer == null) {
-      return false;
-    }
 
-    if (item.stackSize == 1 && !entityPlayer.capabilities.isCreativeMode) {
-      entityPlayer.inventory.setInventorySlotContents(entityPlayer.inventory.currentItem, filledContainer);
-    } else {
-      if (!entityPlayer.capabilities.isCreativeMode) {
-        item.stackSize--;
-        entityPlayer.inventory.setInventorySlotContents(entityPlayer.inventory.currentItem, item);
-      }
-      if (filledContainer.isStackable()) {
+    for (FluidTank subTank : tank.getOutputTanks()) {
+      FluidAndStackResult fill = tryFillContainer(entityPlayer.inventory.getCurrentItem(), subTank.getFluid());
+      if (fill.result.fluidStack != null) {
+
+        if (!entityPlayer.capabilities.isCreativeMode) {
+          subTank.setFluid(fill.remainder.fluidStack);
```

Comment:
It does, the creative-check is around "remove fluid from tank" and "remove empty bucket from player".